src/test/java/it/gov/pagopa/pu/registry/utils/ExtractionUtilsTest.java

```diff
@@ -0,0 +1,38 @@
+package it.gov.pagopa.pu.registry.utils;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ExtractionUtilsTest {
+
+  @Test
+  void testExtractIuds_present() {
+    String str1 = "IUD: id1, id2, id3; CODE: code1, code2;";
+    String str2 = "Hello world. IUD: id4, id5 ; CODE: c1;";
+    String str3 = "Hello. CODE: c2; IUD: id6, id7, id8  ;";
```

Comment:
let's add a test also for a description without the final `;`, actually it is not present (when there is only the IUD)